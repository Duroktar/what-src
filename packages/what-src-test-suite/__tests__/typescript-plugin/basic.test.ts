
import * as ts from 'typescript'
import { createTransformer } from '@what-src/typescript-plugin'

const compilerOptions: ts.CompilerOptions = {
  module: ts.ModuleKind.CommonJS,
  jsx: ts.JsxEmit.React,
  target: ts.ScriptTarget.ESNext,
  importHelpers: true,
}

const transformerOptions = {
  cacheLocOverride: '/CACHE_LOCATION/OVERRIDE',
  createCacheDir: false,
  createCacheFile: false,
}

const sourceText = `
import * as React from 'react'

class TestClassComponent extends React.Component {
  render() {
    return (
      <div className="class-component">
        <p>some p thing</p>
      </div>
    )
  }
}

const TestFunctionalComponent = () => (
  <div className="functional-component">
    <p>some p thing</p>
  </div>
)

const TestMemoizedComponent = React.memo(() => (
  <div className="memoized-component">
    <p>some p thing</p>
  </div>
))
`

it('works', () => {
  const result = ts.transpileModule(sourceText, {
    compilerOptions,
    transformers: { before: [createTransformer(transformerOptions)] },
  })

  expect(result.outputText).toMatchSnapshot()
})
