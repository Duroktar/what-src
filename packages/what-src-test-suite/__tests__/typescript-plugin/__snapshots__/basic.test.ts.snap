// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: true });
require(\\"/CACHE_LOCATION/OVERRIDE/node_modules/@what-src/.cache/runtimeService\\").default;
const React = require(\\"react\\");
class TestClassComponent extends React.Component {
    render() {
        return (React.createElement(\\"div\\", { \\"data-what-src\\": '1' },
            React.createElement(\\"p\\", { \\"data-what-src\\": '2' }, \\"some p thing\\")));
    }
}
const TestFunctionalComponent = () => (React.createElement(\\"div\\", { \\"data-what-src\\": '3' },
    React.createElement(\\"p\\", { \\"data-what-src\\": '4' }, \\"some p thing\\")));
const TestMemoizedComponent = React.memo(() => (React.createElement(\\"div\\", { \\"data-what-src\\": '5' },
    React.createElement(\\"p\\", { \\"data-what-src\\": '6' }, \\"some p thing\\"))));
"
`;
